.github/workflows/build-test.yml

```diff
@@ -16,18 +16,30 @@ jobs:
 
     steps:
       - name: Clone Repo
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
 
       - name: Set up JDK 1.8
         uses: actions/setup-java@v1
         with:
           java-version: 1.8
 
+      - name: Cache Gradle Caches
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches/
+          key: cache-gradle
+
+      - name: Cache Gradle Wrapper
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/wrapper/
+          key: cache-gradle-wrapper
+
       - name: Make gradlew executable
         run: chmod +x ./gradlew
 
       - name: Build project
-        run: ./gradlew build
+        run: ./gradlew clean build
```

Comment:
Why do you need the "clean" since each execution should do a clean checkout, no?